digdag-standards/src/main/java/io/digdag/standards/command/ecs/DefaultEcsClient.java

```diff
@@ -224,13 +249,61 @@ public GetLogEventsResult getLog(
         if (nextToken.isPresent()) {
             request.withNextToken("f/" + nextToken.get());
         }
-        return logs.getLogEvents(request);
+        return retryOnRateLimit(() -> logs.getLogEvents(request));
     }
 
     @Override
     public void close()
             throws IOException
     {
-        client.shutdown();
+        retryOnRateLimit(() -> {
+            client.shutdown();
+            return true; //avoid restrict on generics with void
+        });
+    }
+
+    /**
+     * Retry func if ThrottlingException happen.
+     * Retry interval gradually increases with random jitter.
+     * @param func
+     * @param <T>
+     * @return
+     * @throws AmazonServiceException
+     */
+    @VisibleForTesting
+    public <T> T retryOnRateLimit(Supplier<T> func) throws AmazonServiceException
+    {
+        //ToDo  AmazonECSClient has its own retry policy mechanism. Evaluate it and consider it as replacement of this method.
+        final int baseIncrementalSecs = 10;
+        for (int i = 0; i < rateLimitMaxRetry; i++) {
+            try {
+                return func.get();
+            }
+            catch (AmazonServiceException ex) {
+                if (RetryUtils.isThrottlingException(ex)) {
+                    logger.debug("Rate exceed: {}. Will be retried.", ex.toString());
+                    // Max of baseWaitSecs is rateLimitMaxBaseWaitSecs
+                    final long baseWaitSecs = baseIncrementalSecs * i > rateLimitMaxBaseWaitSecs ? rateLimitMaxBaseWaitSecs: baseIncrementalSecs * i;
```

Comment:
This can be replaced with `Math.min(baseIncrementalSecs * i, rateLimitMaxBaseWaitSecs)`

Reply:
Thanks, I fixed it.